data_model/src/trigger/set.rs

```diff
@@ -40,19 +40,19 @@ pub struct RepeatsOverflowError;
 #[derive(Debug, Default)]
 pub struct Set {
     /// Triggers using [`DataEventFilter`]
-    data_triggers: DashMap<Id, Action<DataEventFilter>>,
+    pub data_triggers: DashMap<Id, Action<DataEventFilter>>,
     /// Triggers using [`PipelineEventFilter`]
-    pipeline_triggers: DashMap<Id, Action<PipelineEventFilter>>,
+    pub pipeline_triggers: DashMap<Id, Action<PipelineEventFilter>>,
     /// Triggers using [`TimeEventFilter`]
-    time_triggers: DashMap<Id, Action<TimeEventFilter>>,
+    pub time_triggers: DashMap<Id, Action<TimeEventFilter>>,
     /// Triggers using [`ExecuteTriggerEventFilter`]
-    by_call_triggers: DashMap<Id, Action<ExecuteTriggerEventFilter>>,
+    pub by_call_triggers: DashMap<Id, Action<ExecuteTriggerEventFilter>>,
     /// Trigger ids with type of events they process
     ids: DashMap<Id, EventType>,
     /// List of actions that should be triggered by events provided by `handle_*` methods.
     /// Vector is used to save the exact triggers order.
     /// Not being cloned
-    matched_ids: RwLock<Vec<(Event, Id)>>,
+    pub matched_ids: Vec<(Event, Id)>,
 }
```

Comment:
We can make them private again once the implementation has moved back inside. I admit this is quite ugly since it is kind of half way towards the new/better way of doing things, aka with patches.